### DISCUSSION TITLE
How to set collection creator as current approved account?

### DISCUSSION THREAD

**Discussion by CryptainGolD** [aptos-labs/aptos-developer-discussions#504] [Questions] [answered]
In NFT-Minting-Dapp template, we define the collection creator address manually, so once I create a collection with account A (another account from the one defineded as `VITE_COLLECTION_CREATOR_ADDRESS `in .env), and mint that collection with account B, then the mint fee does not tranferred to account A, but to `VITE_COLLECTION_CREATOR_ADDRESS `

https://explorer.aptoslabs.com/txn/6173672656/balanceChange?network=testnet

I've defined as follows in `.env`:
`VITE_COLLECTION_CREATOR_ADDRESS="0x9d083d9d3ff91382a0303c6b4860d7b545603695f9868a83230ee3459c384b33"`
But the collection created by `0x2ed57a7ec6a84e6af22105618cc6a2ab79435e36b63b5062c71bf5487ef3d425` on web, and I minted that collection with `0x2ed57a7ec6a84e6af22105618cc6a2ab79435e36b63b5062c71bf5487ef3d425`
then the mint fee must return to `0x2ed57a7ec6a84e6af22105618cc6a2ab79435e36b63b5062c71bf5487ef3d425` but transferred to `0x9d083d9d3ff91382a0303c6b4860d7b545603695f9868a83230ee3459c384b33`

How to make it as I want?

**Comment by 0xbrw** [comment]
If NFT-minting dapp template docs it's recommended that you set the `VITE_COLLECTION_CREATOR_ADDRESS` to be the same as your account address. So account A should be the same as `VITE_COLLECTION_CREATOR_ADDRESS`. 

Source: https://aptos.dev/en/build/create-aptos-dapp/templates/nft-minting-dapp#in-the-env-file-set-vite_collection_creator_address-to-be-the-same-as-your-account-address

**Reply by CryptainGolD** [reply]
I want to change it in the contract please

**Comment by 0xmaayan** [comment]
Not completely sure how you set `VITE_COLLECTION_CREATOR_ADDRESS="0x9d083d9d3ff91382a0303c6b4860d7b545603695f9868a83230ee3459c384b33"` but minted a collection with a different account `0x2ed57a7ec6a84e6af22105618cc6a2ab79435e36b63b5062c71bf5487ef3d425`. The contract would not let you
https://github.com/aptos-labs/create-aptos-dapp/blob/main/templates/nft-minting-dapp-template/contract/sources/launchpad.move#L217-L219


The contract [transfers](https://github.com/aptos-labs/create-aptos-dapp/blob/main/templates/nft-minting-dapp-template/contract/sources/launchpad.move#L380) the mint fee to the [mint_fee_collector_addr](https://github.com/aptos-labs/create-aptos-dapp/blob/main/templates/nft-minting-dapp-template/contract/sources/launchpad.move#L128), which is the account who published the contract. 
This is probably the account that shows up under `/.aptos/config.yaml` file. 

If you want to set the `mint_fee_collector_addr` to be the same as the collection creator address (i.e `VITE_COLLECTION_CREATOR_ADDRESS`), you can set it [here](https://github.com/aptos-labs/create-aptos-dapp/blob/main/templates/nft-minting-dapp-template/contract/sources/launchpad.move#L141) 
```
mint_fee_collector_addr: @initial_creator_addr
```

Note: it depends on *how* you deploy the contract. If you are using the create-aptos-dapp nft minting template flow, you deploy the contract under an object address (which I believe in your case is [this object](https://explorer.aptoslabs.com/object/0xf6eba2fef872c0e10d62bde66dbbc719824a9b2833026e5bbc4beade136aff12/modules/code/LaunchB0xx?network=testnet))
https://github.com/aptos-labs/create-aptos-dapp/blob/main/templates/nft-minting-dapp-template/contract/sources/launchpad.move#L131-L132

**Reply by CryptainGolD** [reply]
Yes, I've already defined like that:
```
    fun init_module(sender: &signer) {
        move_to(sender, Registry {
            collection_objects: vector::empty()
        });
        move_to(sender, Config {
            creator_addr: @initial_creator_addr,
            admin_addr: signer::address_of(sender),
            pending_admin_addr: option::none(),
            // mint_fee_collector_addr: signer::address_of(sender),
            mint_fee_collector_addr: @initial_creator_addr,
        });
    }
```

But this is manual method with entereing creator_address, What I am going to do is, once user create a collection, then set the creator address as that user's account. so when any other user mint that collection, then mint fee goes to the user who created collection. not to creator_address we defined in .env

**Comment by 0x-j** [comment]
Hey hey, in the template the mint fee collector is global per contract, so all mint fee from any collection goes to that addr. You want the mint fee collector to be set at the collection level, right? You can remove it from [global config](https://github.com/aptos-labs/create-aptos-dapp/blob/main/templates/nft-minting-dapp-template/contract/sources/launchpad.move#L128) and add it to [collection config](https://github.com/aptos-labs/create-aptos-dapp/blob/main/templates/nft-minting-dapp-template/contract/sources/launchpad.move#L108), when you create the collection, you can set the fee collector in `CollectionConfig`

**Reply by 0x-j** [reply]
Sounds like you also want anyone to be able to create a collection, we have a version for that.

https://github.com/aptos-labs/move-by-examples/blob/main/nft-launchpad/aptos/move/sources/launchpad.move

live deployment https://aptos-nft-launchpad.vercel.app/

**Reply by CryptainGolD** [reply]
Yes, but this version will transfers the mint fee to a special defined address.
It means, once I create a collection with my wallet, and another user mint my collection, then the mint fee that I defined when creating collection, do not to come to my wallet. it's something wrong I think.
Does it must come in my wallet, doesn't it? my wallet is the one created that collection, but fee goes to somewhere or a specific address defined in .env.
Please make me sense for this.
What I am going to do is, the collection creator get some percentage of mint fee and the mint fee collector get rest of that.

**Reply by 0x-j** [reply, marked as answer]
>  the collection creator get some percentage of mint fee and the mint fee collector get rest of that.

You can define how the fee is splitted in the CollectionConfig, you just put 2 addresses there, one is collection creator and the other is mint fee collector. 

```
    struct CollectionConfig has key {
        // Key is stage, value is mint fee denomination
        mint_fee_per_nft_by_stages: SimpleMap<String, u64>,
        mint_enabled: bool,
        collection_owner_obj: Object<CollectionOwnerObjConfig>,
        extend_ref: object::ExtendRef,
        mint_fee_collector: address,
        collection_creator: address,
    }
```
